### ISSUE TITLE
logLevel for custom levels

### ISSUE THREAD

**Issue by nemisj** [pinojs/pino#177]
Didn't know where to ask, that's why asking here.

I wanted to use custom level, but in my stream, even when i log using the custom method i get the `level` which is currently set and not the custom one.

I've found that it's get set in the code:

```javascript
this[key] = isStandardLevel(key) ? Pino.prototype[key] : genLog(num)
```

Is this a desired behaviour or can we change it to real level of custom? into something like:

```javascript
this[key] = isStandardLevel(key) ? Pino.prototype[key] : genLog(this.levels.values[key])
```

**Comment by davidmarkclements** [comment]
That's for when `levelVal` is being set - `genLog(num)` is generating a custom logger function

It does sound like a bug to me - can you show some code w/ output?

**Comment by nemisj** [comment]
```javascript
const pino = require('pino');

const stream = {
  write(r) {
		console.log(r);
  }
};

const log = pino({
  name: 'app',
  safe: true
}, stream);

log.addLevel('zork', 25);

log.level = 'debug';

log.zork('hohoho');
```

will output:

`{"pid":2791,"hostname":"deller","name":"app","level":20,"time":1484755820829,"msg":"hohoho","v":1}`

which has level: 20 instead of 25

**Comment by jsumners** [comment]
Clearly I did a wonderful job of implementing this feature.

I'll get it resolved.

**Comment by mcollina** [comment]
@jsumners the actual reason is that I saw it was a _very complex_ feature, and I didn't implement them on the first go. Thanks for helping out :).

**Comment by jsumners** [comment]
This is partially affected by issue #167 .

**Comment by jsumners** [comment]
The fix for this is in in PR https://github.com/pinojs/pino/pull/168 . I opted to put it in that PR because testing for this issue was affected by the shared state problem.

**Comment by jsumners** [comment]
Fixed by #168

**Comment by nemisj** [comment]
Thanks

**Comment by jsumners** [comment]
It's published as 3.1.3
